chapters/synchronization.txt

```diff
@@ -53,67 +53,133 @@ _memory dependencies_ between two sets of operations defined by the
 command's two _synchronization scopes_.
 
 [[synchronization-dependencies-scopes]]
-The synchronization scopes define which other operations a synchronization
-command is able to create execution dependencies with.
-Any type of operation that is not in a synchronization command's
+A synchronization scope is a set of selection criteria for operations.
+A synchronization command has a first synchronization scope and a second
+synchronization scope for the purpose od declaring which kinds of operations
+are subject to the dependency introduced by the command.
+Any type of operation that is not selected by the synchronization command's
 synchronization scopes will not be included in the resulting dependency.
+
+[NOTE]
+.Note
+====
 For example, for many synchronization commands, the synchronization scopes
-can: be limited to just operations executing in specific
+can: be limited to just operations executing in only specific
 <<synchronization-pipeline-stages,pipeline stages>>, which allows other
 pipeline stages to be excluded from a dependency.
-Other scoping options are possible, depending on the particular command.
+But other scope to other kinds of operations is possible depending on the
+particular command, such as the <<synchronization-fences-signaling,
+fence signal operation>>.
+====
 
 [[synchronization-dependencies-execution]]
-An _execution dependency_ is a guarantee that for two sets of operations,
-the first set must: _happen-before_ the second set.
-If an operation happens-before another operation, then the first operation
-must: complete before the second operation is initiated.
+An _execution dependency_ is an implementation's guarantee that for two sets
+of operations, the first set must: _happen-before_ the second set.
+If an operation _happens-before_ another operation, it means the first
+operation must: complete before the second operation is initiated.
 More precisely:
 
-  * Let *A* and *B* be separate sets of operations.
+  * Let *Ops* be a set of all submitted operations.
   * Let *S* be a synchronization command.
-  * Let *A~S~* and *B~S~* be the synchronization scopes of *S*.
-  * Let *A'* be the intersection of sets *A* and *A~S~*.
-  * Let *B'* be the intersection of sets *B* and *B~S~*.
-  * Submitting *A*, *S* and *B* for execution, in that order, will result in
-    execution dependency *E* between *A'* and *B'*.
-  * Execution dependency *E* guarantees that *A'* happens-before *B'*.
+  * Let *Scope~src~* be the first synchronization scope of *S*.
+  * Let *Scope~dst~* be the second synchronization scope of *S*.
+  * Let *Ops~src~* be the subset of operations in *Ops* selected by scope
+    *Src~S~*.
```

Comment:
Should this be Scope~Src~ not Src~S~? And similarly for the destination scope below?

Reply:
Yea. 

Going back and forth on those var names. Suggestions?

Reply:
Hmm - Scope~src makes sense to me. I think that may be better than Src S because in some places in the spec the variable **S** is used to mean a synchronization command (rather than a scope).

Other people may have more ideas though!

Reply:
\*Scope\~src~* is fine, though I'd be  tempted to call it \*Scope\~first~* (and second) to match the wording in the spec.

Reply:
@Tobski The nomenclature is sorta mixed. The API parameters use src/dst. TBH I like it more because first/second does not convey a difference between the scope in how they are used.

Reply:
We landed on first/second for some reason that I honestly can't entirely remember. I'm fine with src/dst here, but might be worth considering a nomenclature change for the spec elsewhere to match - though I'd do that in a separate MR because that's going to be churny.